File: core-arch/src/main/java/com/huyingbao/core/arch/module/RxFluxModule.java
package com.huyingbao.core.arch.module;

import android.app.Application;
import android.content.Context;

import androidx.lifecycle.ViewModelProvider;

import com.huyingbao.core.arch.store.RxStoreFactory;

import javax.inject.Singleton;

import dagger.Binds;
import dagger.Module;

/**
 * 依赖注入仓库，提供 1.Application， 2.{@link ViewModelProvider.Factory}实现类{@link RxStoreFactory}
 * <p>
 * Created by liujunfeng on 2019/1/1.
 */
@Module
public abstract class RxFluxModule {
    @Singleton
    @Binds
    abstract Context bindApplication(Application application);

    @Singleton
    @Binds
    abstract ViewModelProvider.Factory bindViewModelFactory(RxStoreFactory rxStoreFactory);
}

File: core-arch/src/main/java/com/huyingbao/core/arch/module/RxFluxModule.kt
package com.huyingbao.core.arch.module

import com.huyingbao.core.arch.action.FlowActionManager
import com.huyingbao.core.arch.dispatcher.RxDispatcher
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import javax.inject.Singleton


/**
 * 依赖注入仓库
 *
 * 1.提供[FlowActionManager]
 *
 * 2.提供[RxDispatcher]
 *
 * Created by liujunfeng on 2019/1/1.
 */
@Module
@InstallIn(ApplicationComponent::class)
class RxFluxModule {
    /**
     * 提供[FlowActionManager]
     */
    @Singleton
    @Provides
    fun provideFlowActionManager(): FlowActionManager {
        return FlowActionManager()
    }

    /**
     * 提供[RxDispatcher]
     */
    @Singleton
    @Provides
    fun provideRxDispatcher(): RxDispatcher {
        return RxDispatcher()
    }
}
